src/shared/compounders/AMMCompounderBase.sol

```diff
@@ -164,6 +167,17 @@ abstract contract AMMCompounderBase is ReentrancyGuard, ERC4626 {
         return totalAUM;
     }
 
+    /// @dev Returns the maximum amount of the underlying asset that can be deposited into the Vault for the receiver, through a deposit call.
+    function maxDeposit(address) public view override returns (uint256) {
+        uint256 _assetCap = convertToAssets(depositCap);
+        return _assetCap == 0 ? type(uint256).max : _assetCap - totalAUM;
+    }
+
+    /// @dev Returns the maximum amount of the Vault shares that can be minted for the receiver, through a mint call.
+    function maxMint(address) public view override returns (uint256) {
+        return depositCap == 0 ? type(uint256).max : depositCap - totalSupply;
```

Comment:
Maybe for my understanding, what is the difference between `totalAUM` and `totalSupply`? The total supply of what does the latter refer to?

Reply:
- `totalAUM` refer to the total amount of _assets_ managed by the vault.
- `totalSupply` refers to the amount of shares issued (a standard ERC20 variable).

Reply:
Wasn't that meant to be a 1:1 relationship?

Reply:
In the Concentrator they maintain a 1:1 relationship, in the Compounders each share represents more and more assets as harvests happen.